core/src/main/scala/kafka/server/metadata/KRaftMetadataCache.scala

```diff
@@ -63,6 +63,7 @@ class KRaftMetadataCache(val brokerId: Int) extends MetadataCache with Logging w
                                        brokers: Array[Int],
                                        listenerName: ListenerName,
                                        filterUnavailableEndpoints: Boolean): java.util.List[Integer] = {
+    // TODO KAFKA-15362
```

Comment:
Do we know what the perf impact on metadata requests is by adding the logic of KAFKA-15362 here? Especially with the warning comment in [L60](https://github.com/apache/kafka/blob/4612fe42af0df0a4c1affaf66c55d01eb6267ce3/core/src/main/scala/kafka/server/metadata/KRaftMetadataCache.scala#L60) regard adding extra logic to `KRaftMetadataCache.maybeFilterAliveReplicas`.